sunpy/util/decorators.py

```diff
@@ -174,10 +196,12 @@ def deprecate(obj, message=message, name=name, alternative=alternative):
             'obj_type': obj_type_name})) +
             altmessage)
 
+        message += removal_message
```

Comment:
I suggest changing this.  As is, it appends `removal_message` if `pending=False` (so it confusingly juxtaposes "may be removed in a future version" and "scheduled for removal in ...") and can't be avoided with a custom message.  Instead, I think `removal_message` should be appended only if `pending=True` and `message=None` (i.e., add at line ~15).

Reply:
Ah good point. Though this means that if you specify `message` you don't get the automatic removal version calculation.

Reply:
I think you still want to see `removal_message` even if `pending=False` though. This is more a question of wording I think. For both pending or not pending, could we substitute "future version" with the calculated removal version?

Reply:
> Though this means that if you specify `message` you don't get the automatic removal version calculation.

If we have something that needs a long deprecation cycle (e.g., `Map` changes), there needs to be a way to show a custom message without the smart-version sentence.  It shouldn't say, e.g., "This function will be removed in version 4.1.  This is scheduled for removal in 3.1."

> For both pending or not pending, could we substitute "future version" with the calculated removal version?

That's probably the easiest fix.

Reply:
So I've now used our smart versioning to calculate a `removal_version` and used this when `pending=False`. When `pending=True`, I've just used `since` as this is version that the function will be deprecated, not removed.

Reply:
Oh, so "pending" _is_ supposed to mean "deprecation is pending", not "removal is pending"?  Okay

Reply:
Yes, I believe this is the astropy deprecated decorator definition of "pending"